langchain4j-mongodb-atlas/src/test/java/dev/langchain4j/store/embedding/mongodb/MongoDbEmbeddingStoreLegacyIndexesIT.java

```diff
@@ -0,0 +1,74 @@
+package dev.langchain4j.store.embedding.mongodb;
+
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoCollection;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.model.embedding.EmbeddingModel;
+import dev.langchain4j.store.embedding.EmbeddingStore;
+import dev.langchain4j.store.embedding.EmbeddingStoreIT;
+import org.bson.Document;
+import org.junit.jupiter.api.AfterEach;
+import org.testcontainers.shaded.com.google.common.collect.Sets;
+
+import java.util.Set;
+
+import static dev.langchain4j.store.embedding.mongodb.MongoDbTestFixture.EMBEDDING_MODEL;
+import static dev.langchain4j.store.embedding.mongodb.MongoDbTestFixture.createClientFromEnv;
+
+public class MongoDbEmbeddingStoreLegacyIndexesIT extends EmbeddingStoreIT {
+
+    private final MongoDbTestFixture helper = initializeTestHelper();
+
+    private MongoDbTestFixture initializeTestHelper() {
+        MongoDbTestFixture helper = new MongoDbTestFixture(createClient());
+        IndexMapping indexMapping = IndexMapping.builder()
+                .dimension(EMBEDDING_MODEL.dimension())
+                .metadataFieldNames(Sets.newHashSet("test-key"))
+                .build();
+        helper.getDatabase().createCollection("test_collection");
+        MongoCollection<Document> collection = helper.getDatabase().getCollection("test_collection");
+        collection.createSearchIndex("test_index_legacy", fromIndexMappingLegacyKnnVector(indexMapping));
+        return helper.initialize(builder -> builder.createIndex(false).indexName("test_index_legacy"));
```

Comment:
@Martin7-1 :
> Also, it's a breaking change, maybe we need documents to tell users how to migrate it from AtlasSearch to AtlasVectorSearch in detail.

I believe this is not a breaking change. The tests here create the old index (I have appended `_legacy` for clarity, though I may need to revert this for M0 tier testing), but they still pass when using the updated aggregation stage.

I agree the docs should include information about upgrading, though this might be as simple as telling users to delete their old index (and re-create manually if needed).

Reply:
OK. got it.